parlai/mturk/webapp/server.py

```diff
@@ -482,6 +496,85 @@ def start_server(port=DEFAULT_PORT, hostname=DEFAULT_HOSTNAME,
     tornado.ioloop.IOLoop.current().start()
 
 
+def crawl_dir_for_tasks(search_dir):
+    found_dirs = {}  # taskname -> task directory
+    contents = os.listdir(search_dir)
+    for sub_dir in contents:
+        full_sub_dir = os.path.join(search_dir, sub_dir)
+        if os.path.exists(os.path.join(full_sub_dir, 'run.py')):
+            # assume this is a task
+            found_dirs[sub_dir] = full_sub_dir
+        if os.path.isdir(full_sub_dir):
+            found_dirs.update(crawl_dir_for_tasks(full_sub_dir))
+    return found_dirs
+
+def rebuild_source():
+    # TODO move task parsing to its own helper file
+    # Copy files to the correct locations,
+    if os.path.exists('dev/task_components'):
+        print('removing old build')
+        sh.rm(shlex.split('-rf {}'.format('dev/task_components')))
+    os.mkdir('dev/task_components')
+
+    copy_dirs = {}
+    parlai_task_dir = os.path.join(parlai_path, 'mturk', 'tasks')
+    tasks.update(crawl_dir_for_tasks(parlai_task_dir))
+
+    if parlai_int_path is not None:
+        parlai_internal_task_dir = \
+            os.path.join(parlai_int_path, 'mturk', 'tasks')
+        tasks.update(crawl_dir_for_tasks(parlai_internal_task_dir))
+
+    print (tasks)
```

Comment:
nit: necessary print statement?

Reply:
Necessary for my debugging in the next step, but on the chopping block to be removed after